src/main/java/org/datadog/jenkins/plugins/datadog/listeners/DatadogItemListener.java

```diff
@@ -0,0 +1,131 @@
+package org.datadog.jenkins.plugins.datadog.listeners;
+
+import hudson.Extension;
+import hudson.model.Item;
+import hudson.model.listeners.ItemListener;
+import org.datadog.jenkins.plugins.datadog.DatadogClient;
+import org.datadog.jenkins.plugins.datadog.DatadogEvent;
+import org.datadog.jenkins.plugins.datadog.DatadogUtilities;
+import org.datadog.jenkins.plugins.datadog.events.ItemCRUDEventImpl;
+import org.datadog.jenkins.plugins.datadog.events.ItemCopiedEventImpl;
+import org.datadog.jenkins.plugins.datadog.events.ItemLocationChangedEventImpl;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Logger;
+
+/**
+ * This class registers an {@link ItemListener} to trigger events and calculate metrics:
+ * - When an item gets created, the {@link #onCreated(Item)} method will be invoked.
+ * - When an item gets copied, the {@link #onCopied(Item, Item)} method will be invoked.
+ * - When an item gets deleted, the {@link #onDeleted(Item)} method will be invoked.
+ * - When an item gets updated, the {@link #onUpdated(Item)} method will be invoked.
+ * - When an item gets their location changed, the {@link #onLocationChanged(Item, String, String)} method will be invoked.
+ */
+@Extension
+public class DatadogItemListener extends ItemListener {
+
+    private static final Logger logger = Logger.getLogger(DatadogItemListener.class.getName());
+
+    @Override
+    public void onCreated(Item item) {
+        onCRUD(item, ItemCRUDEventImpl.CREATED);
+    }
+
+    @Override
+    public void onDeleted(Item item) {
+        onCRUD(item, ItemCRUDEventImpl.DELETED);
+    }
+
+    @Override
+    public void onUpdated(Item item) {
+        onCRUD(item, ItemCRUDEventImpl.UPDATED);
+    }
+
+    private void onCRUD(Item item, String action) {
+        try {
+            final boolean emitSystemEvents = DatadogUtilities.getDatadogGlobalDescriptor().isEmitSystemEvents();
+            if (!emitSystemEvents) {
+                return;
+            }
```

Comment:
Is there a way to have this check in a top-level place and avoid having this once per Listener method ?
Like at instantiation for example, and we wouldn't even register the listeners at all ?

Reply:
Since i use `return`, i don't think so. I could create an exception but i will still have to handle it when it would be thrown.